download_logs.py

```diff
@@ -20,41 +23,101 @@ def get_arguments():
+    error_ids = [error_id_table[error_label] for error_label in error_labels] # creates list of error_ids
+    return error_ids
+
+
 def main():
     args = get_arguments()
 
     try:
         # the db_info_api sends a json file with a list of all public database entries
-        db_entries_list = requests.get(url=args.db_info_api).json()
+        db_entries_list = requests.get(url=args.db_info_api).json() # copy list
     except:
         print("Server request failed.")
         raise
 
     if args.print_entries:
-        # only print the json output
+        # only print the json output without downloading
         print(db_entries_list)
+
     else:
-        if not os.path.isdir(args.download_folder):
+        if not os.path.isdir(args.download_folder): # returns true if path is an existing directory
             print("creating download directory " + args.download_folder)
-            os.makedirs(args.download_folder)
+            os.makedirs(args.download_folder) # creates new folder
         # find already existing logs in download folder
         logfile_pattern = os.path.join(os.path.abspath(args.download_folder), "*.ulg")
         logfiles = glob.glob(os.path.join(os.getcwd(), logfile_pattern))
         logids = frozenset(os.path.splitext(os.path.basename(f))[0] for f in logfiles)
 
+        # filter for mav types
+        if args.mav_type is not None:
+            mav = [mav_type.lower() for mav_type in args.mav_type]
+            db_entries_list = [entry for entry in db_entries_list if set([entry["mav_type"].lower()]).issubset(set(mav))]
+
+        # filter for rating
+        if args.rating is not None:
+            rate = [rating.lower() for rating in args.rating]
+            db_entries_list = [entry for entry in db_entries_list if set([entry["rating"].lower()]).issubset(set(rate))]
+
+        # filter for error labels
+        if args.error_labels is not None:
+            err_labels = error_labels_to_ids(args.error_labels)
+            db_entries_list = [entry for entry in db_entries_list if set(err_labels).issubset(set(entry["error_labels"]))]
+            # compares numbers, must contain all
+
+        # filter for flight modes
+        if args.flight_modes is not None:
+            modes = flight_modes_to_ids(args.flight_modes)
+            db_entries_list = [entry for entry in db_entries_list if set(modes).issubset(set(entry["flight_modes"]))]
+            # compares numbers, must contain all
+
         # set number of files to download
-        n_en = len(db_entries_list)
-        if (args.max_num > 0):
-            n_en = min(n_en, args.max_num)
+        n_en = len(db_entries_list) # max number of logs
+        n = 0  # min number for iteration
+        if args.max_num > 0: # default is -1
+            n = n_en - min(n_en, args.max_num) # difference
+
+        # reverse list order to first download newest log files
+        db_entries_list = [db_entries_list[e] for e in range(n_en-1, n-1, -1)]
```

Comment:
How about `db_entries_list = reversed(db_entries_list)`?
However we don't have any guarantee that the server gives us the items in a specific order. So either you need to make sure the server orders the items, or you order by date here (I slightly prefer to do it here).